Updated xcube Dataset Specification (#850)
dcs4cop/xcube: docs/source/cubespec.md

_may_ is optional and recommended in cases (like good to have).

## Table of Contents

- [Data Model and Format](#Data Model and Format)
- [Metadata](#Metadata)
- [Spatial Reference](#Spatial Reference)
- [Temporal Reference](#Temporal Reference)
- [Encoding of Units](#Encoding of Units)
- [Encoding of Flags](#Encoding of Flags)
- [Encoding of Missing Values](#Encoding of Missing Values)
- [Encoding of Scale/Offset Values](#Encoding of Scale/Offset Values)
- [Multi-Resolution Datasets](#Multi-Resolution Datasets)
- [Multi-Band Datasets](#Multi-Band Datasets)
- [Metadata Consolidation](#Metadata Consolidation)
- [Zip Archives](#Zip Archives)
- [Dataset Examples](#Dataset Examples)

## Data Model and Format

The data model for xcube gridded datasets is a subset of the 
[Unidata Common Data Model] (CDM). The CDM has originally been established for 
NetCDF (Network Common Data Form) but is implemented also for the OPenDAP 
protocol and the HDF and Zarr formats. Other common gridded data models can be 
easily translated into the CDM, for example GeoTIFF.

xcube gridded datasets must fully comply to the _Climate and Forecast

> Metadata Conventions_, short [CF Conventions]. The following sections in this 

Reviewer: ```suggestion
Metadata Conventions_, [CF Conventions] for short. The following sections in this 
```

chapter describe how these conventions are tailored and applied to xcube 
datasets. By default, xcube gridded datasets are made available in the 
[Zarr format v2] as this format is ideal for Cloud storage, e.g. Object 

> Storage such as AWS S3 and can be easily converted to other formats such as 

Reviewer: ```suggestion
Storage such as AWS S3, and can be easily converted to other formats such as 
```

NetCDF or TIFF. We consider the popular [xarray] Python package to be the 
primary in-memory representation for xcube gridded datasets. The 
[xarray.Dataset] data model represents a subset of the CDM, interprets CF 
compliant data correctly, and we will follow this simplified CDM model for 
xcube gridded datasets in both structure and terminology:

[Unidata Common Data Model]: https://www.unidata.ucar.edu/software/netcdf-java/v4.6/CDM/
[CF Conventions]: https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html
[Zarr format v2]: https://zarr.readthedocs.io/en/stable/spec/v2.html
[xarray]: http://xarray.pydata.org/
[xarray.Dataset]: http://xarray.pydata.org/en/stable/user-guide/data-structures.html#dataset

* A **dataset** is a container for _variables_. A dataset has metadata in form 
  of a key-value mapping (global attributes).
* A **variable** has N dimensions in a prescribed order, and each dimension 

>   has a name and a size. A variable contains the actual gridded data in form 

Reviewer: ```suggestion
  has a name and a size. A variable contains the actual gridded data in the form 
```

>   of an N-D data array that has the shape of the dimension sizes. The 

Reviewer: ```suggestion
  of an N-dimensional data array that has the shape of the dimension sizes. The 
```

  data type of the array should be numeric, however also text strings and 
  complex types are allowed. A variable has metadata in form of a key-value 
  mapping (variable attributes). There are two types of variables, 
  _coordinates_ and _data variables_.
* **Coordinates** usually contain the labels for a dimension. For example, 

>   the coordinate `lon` has a 1-D array that contains 3600 longitude values for 
>   the dimension `lon` of size 3600. However, coordinates may also be 2-D. 

Reviewer: ```suggestion
  the coordinate `lon` could have a 1-D array that contains 3600 longitude values for 
  the dimension `lon` of size 3600. However, coordinates may also be 2-D. 
```

  For example. a dataset using satellite geometry may provide its `lon` 

>   coordinate as a 2-D array for the dimensions `x` and `y` (for which should 
>   exists coordinates too).

Reviewer: ```suggestion
  coordinate as a 2-D array for the dimensions `x` and `y` (for which co-ordinates
  should also exist).
```

* All variables that are not coordinates are **data variables**. 
  They provide the actual dataset’s data. For each of the dimensions used by a 
  data variable, a corresponding coordinate must exist. For example, for a 
  data variable `NDVI` with dimensions `time`, `lat`, `lon`, the coordinates
  `time`, `lat`, `lon` must exist too.

## Metadata

> Global and variable metadata shall follow the recommendations of the 

Reviewer: ```suggestion
Global and variable metadata must follow the recommendations of the 
```

[ESIP Attribute Convention for Data Discovery v1.3]. If other metadata 
attributes are used, they should be covered by the [CF Conventions v1.8].

[CF Conventions v1.8]: https://cfconventions.org/Data/cf-conventions/cf-conventions-1.8/cf-conventions.html
[ESIP Attribute Convention for Data Discovery v1.3]: https://wiki.esipfed.org/Attribute_Convention_for_Data_Discovery_1-3

## Spatial Reference

The spatial dimensions of a data variable must be the innermost dimensions, 
namely `lat`, `lon` for geographic (EPGS:4326, WGS-84) grids, and `y`, `x` 
for other grids – in this order.
Datasets that use a geographic (EPGS:4326, WGS-84) grid must provide the 
1-D coordinates `lat` for dimension `lat` and `lon` for dimension `lon`.
Datasets that use a non-geographic grid must provide the 1-D coordinates 
`y` for dimension `y` and `x` for dimension `x`. 

* In case the grid refers to a known spatial reference system (projected CRS), 

>   it must make use of the CRS encoding described in the 

Reviewer: ```suggestion
  the dataset must make use of the CRS encoding described in the 
```

  [CF Conventions on Grid Mapping], i.e. add a variable `crs`.

> * In case the grid is referring to satellite viewing geometry must provide 

Reviewer: ```suggestion
* In case the grid is referring to satellite viewing geometry, the dataset must provide 
```
